### PR TITLE
Humble release versions 2024-01-26

### PR DESCRIPTION
**Pull request by audrow**
- [x] Merge in https://github.com/ros2/rosbag2/pull/1551

Note, this is actually patch release 7. I misremembered when creating the branch. I can change it if it'd be good to do.

---

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/20136/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/14574/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/20847/)
* RHEL [Build Status](https://ci.ros2.org/job/ci_linux-rhel/562/)

https://ci.ros2.org/view/packaging/job/packaging_linux/3313/
https://ci.ros2.org/view/packaging/job/packaging_linux-aarch64/2671/
https://ci.ros2.org/view/packaging/job/packaging_linux-rhel/1815/
https://ci.ros2.org/view/packaging/job/packaging_windows/3126/
https://ci.ros2.org/view/packaging/job/packaging_windows_debug/2086/

### CODE REVIEW

**Comment by clalancette** [comment]
> Note, this is actually patch release 7. I misremembered when creating the branch. I can change it if it'd be good to do.

Nah, not necessary.  This is a temporary branch anyway and will be deleted.

File: ros2.repos [hunk 1]
```diff
@@ -42,11 +42,11 @@ repositories:
   eclipse-cyclonedds/cyclonedds:
     type: git
     url: https://github.com/eclipse-cyclonedds/cyclonedds.git
-    version: 0.10.3
+    version: 0.10.4
```
Comment (clalancette) [inline ros2.repos:45]: I think it is reasonable to do this update, but in that case we should also `bloom-release` it to 0.10.4.

File: ros2.repos [hunk 2]
```diff
@@ -58,15 +58,15 @@ repositories:
   osrf/osrf_pycommon:
     type: git
     url: https://github.com/osrf/osrf_pycommon.git
-    version: 2.0.2
+    version: 2.1.4
```
Comment (clalancette) [inline ros2.repos:61]: Same thing here; this seems like a reasonable update, but if we are going to do this we should also do a `bloom-release` of it into Humble.

**Review by clalancette — CHANGES_REQUESTED** [review]

**Comment by clalancette** [comment]
The version bumps look good to me, but the RHEL compile failure is odd.  Either we broke something in the CI scripts (we don't test building on RHEL-8 very often), or we broke something in the Humble code.  Either way, we should take a look and see if we can fix it.

**Comment by clalancette** [comment]
> The version bumps look good to me, but the RHEL compile failure is odd. Either we broke something in the CI scripts (we don't test building on RHEL-8 very often), or we broke something in the Humble code. Either way, we should take a look and see if we can fix it.

Ah, I see what is happening.  It looks like we are installing Python 3.11 in the RHEL-8 container now, and when we do `FindPython` in CMake, we are finding that.  Because `catkin_pkg` is only installed for Python 3.6, we can't find it.

@cottsay Any ideas on what is going on here, and possible thoughts for a solution?

**Comment by clalancette** [comment]
OK, one more piece of information.  It looks like mesa-libGL-devel 23.1.4 (which we explicitly install [here](https://github.com/ros2/ci/blob/403b3445ee7e7253dd6c24b042feb3ae584dce76/linux_docker_resources/Dockerfile-RHEL#L110)) now requires python3.11, which is why we end up with it installed.  Maybe we either need to not install that (though I don't know the consequences), or possibly force RHEL-8 to continue to use python3.6 for everything (I'm not sure how that would work).

**Comment by cottsay** [comment]
> possibly force RHEL-8 to continue to use python3.6 for everything

I do this on Fedora all the time. There are CMake flags to do it. I'll open a PR and we'll give it a shot.

**Comment by cottsay** [comment]
The magic flag is `-DPython3_EXECUTABLE:STRING=/usr/bin/python3`.

[Build Status](https://ci.ros2.org/job/ci_linux-rhel/562/)

We should consider treating this like the C++ version and setting the default CMake value somewhere central. I think we want to use the system "default" python version on pretty much all supported platforms and not necessarily the "newest" version.

Certainly not suggesting we don't provide a mechanism to override, but defaulting to system default seems appropriate, and anchoring off from whichever version provides the `python3` executable seems like a safe bet.

**Review by clalancette — APPROVED** [review]
With the latest fixes to CI (which I just merged), this looks good to me.  Thanks @audrow and @cottsay for looking into it.

**Pull request merged**